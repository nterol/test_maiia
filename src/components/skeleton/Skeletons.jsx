import React from "react";
import { AnimatePresence, motion } from "framer-motion";

import styles from "../styles/layout.module.scss";
import SkeletonArticle from "./SkeletonArticle";

const Skeletons = ({ isLoading }) => (
  <AnimatePresence>
    {isLoading && (
      <motion.section exit={{y: 100}} className={styles.grid}>
        <SkeletonArticle simple={3} duration={0.9} />
        <SkeletonArticle simple={4} duration={0.7} />
        <SkeletonArticle simple={1} duration={0.5} />
        <SkeletonArticle simple={6} duration={0.3} />
        <SkeletonArticle simple={5} duration={0.2} />
      </motion.section>
    )}
  </AnimatePresence>
);

export default Skeletons;
