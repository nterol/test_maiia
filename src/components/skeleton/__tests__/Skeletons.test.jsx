import React from "react";
import renderer from "react-test-renderer";

import Skeletons from "../Skeletons";

it("should match snapshot", () => {
  const snap = renderer.create(<Skeletons isLoading/>);

  expect(snap).toMatchSnapshot();
});
