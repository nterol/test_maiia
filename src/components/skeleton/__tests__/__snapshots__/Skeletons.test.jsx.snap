// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<section
  className="grid"
  style={Object {}}
>
  <article
    className="article"
    style={
      Object {
        "opacity": 0,
        "transform": "translateY(-100px) translateZ(0)",
      }
    }
  >
    <hr
      className="thumbnail"
    />
    <hr
      className="hr"
    />
    <hr
      className="hr"
    />
    <hr
      className="hr"
    />
  </article>
  <article
    className="article"
    style={
      Object {
        "opacity": 0,
        "transform": "translateY(-100px) translateZ(0)",
      }
    }
  >
    <hr
      className="thumbnail"
    />
    <hr
      className="hr"
    />
    <hr
      className="hr"
    />
    <hr
      className="hr"
    />
    <hr
      className="hr"
    />
  </article>
  <article
    className="article"
    style={
      Object {
        "opacity": 0,
        "transform": "translateY(-100px) translateZ(0)",
      }
    }
  >
    <hr
      className="thumbnail"
    />
    <hr
      className="hr"
    />
  </article>
  <article
    className="article"
    style={
      Object {
        "opacity": 0,
        "transform": "translateY(-100px) translateZ(0)",
      }
    }
  >
    <hr
      className="thumbnail"
    />
    <hr
      className="hr"
    />
    <hr
      className="hr"
    />
    <hr
      className="hr"
    />
    <hr
      className="hr"
    />
    <hr
      className="hr"
    />
    <hr
      className="hr"
    />
  </article>
  <article
    className="article"
    style={
      Object {
        "opacity": 0,
        "transform": "translateY(-100px) translateZ(0)",
      }
    }
  >
    <hr
      className="thumbnail"
    />
    <hr
      className="hr"
    />
    <hr
      className="hr"
    />
    <hr
      className="hr"
    />
    <hr
      className="hr"
    />
    <hr
      className="hr"
    />
  </article>
</section>
`;
